### PR TITLE
Marc Alzapiedi

### PR DESCRIPTION
**Pull request by marc-alzapiedi**

### CODE REVIEW

File: spec/cohortManager.spec.js [hunk 1]
```diff
@@ -0,0 +1,441 @@
...
+      }
+    ]
+    // Execute
+    cohortManager.create('Cohort 9')
+    const result = cohortManager.create('Cohort 10')
+    // Verify
+    expect(result).toEqual(expected)
+  })
+
+  // Test 2
```
Comment (sarajbeazley) [inline spec/cohortManager.spec.js:30]: Was this one of the requirements?

File: spec/cohortManager.spec.js [hunk 2]
```diff
@@ -0,0 +1,441 @@
...
+    cohortManager.create('Cohort 8')
+    cohortManager.create('Cohort 9')
+    cohortManager.create('Cohort 10')
+    const result = cohortManager.searchCohort('Cohort 8')
+
+    // Verify
+    expect(result).toEqual(expected)
+  })
+
+  // Test 3
```
Comment (sarajbeazley) [inline spec/cohortManager.spec.js:49]: simple, neat way of throwing the error message

File: spec/cohortManager.spec.js [hunk 3]
```diff
@@ -0,0 +1,441 @@
...
+    cohortManager.create('Cohort 9')
+    cohortManager.create('Cohort 10')
+
+    // verify
+    expect(() => cohortManager.searchCohort(cohort)).toThrowError(
+      "The cohort doesn't exist"
+    )
+  })
+
+  // Test 4
```
Comment (sarajbeazley) [inline spec/cohortManager.spec.js:66]: does this test take into account adding a student to a specific cohort?

File: src/cohortManager.js [hunk 4]
```diff
@@ -0,0 +1,84 @@
+class CohortManager {
+  constructor() {
+    this.cohortData = []
+  }
+
+  create(cohort) {
```
Comment (sarajbeazley) [inline src/cohortManager.js:6]: good use of the variable name obj

File: src/cohortManager.js [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+      return obj.cohortName === cohort
+    })
+
+    if (searchResult[0] === undefined) {
+      throw new Error("The cohort doesn't exist")
+    }
+    return searchResult[0]
+  }
+
+  addStudent(firstName, lastName, githubUsername, email, cohort) {
```
Comment (sarajbeazley) [inline src/cohortManager.js:27]: did you need all the student info? would the ID alone not have been enough to identify each student (provided all ID's were unique numbers)

File: src/cohortManager.js [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+      this.cohortData.forEach((obj) => {
+        if (obj.students.indexOf(this.searchStudent(first, last)) === -1) {
+          return
+        }
+        obj.students.splice(
+          obj.students.indexOf(this.searchStudent(first, last)),
+          1
+        )
+      })
+    } catch (err) {
```
Comment (sarajbeazley) [inline src/cohortManager.js:85]: I'm not familiar with this catch method - need to look at that

File: src/cohortManager.js [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+    })
+    if (searchResult[0] === undefined) {
+      throw new Error("The student doesn't exist")
+    }
+    return searchResult[0]
+  }
+
+  removeStudent(first, last) {
+    this.searchStudent(first, last)
+    try {
```
Comment (sarajbeazley) [inline src/cohortManager.js:74]: I'm not familiar with this try method, need to look at that

**Review by sarajbeazley — COMMENTED** [review]
Great work Marc!

File: spec/cohortManager.spec.js (hunk 3 above)
Comment (marc-alzapiedi) [reply spec/cohortManager.spec.js:66]: The way I have structured my local datastore is:

```
this.cohortData = [
      {
        cohortName: 'Cohort 9',
        students: [
          {
            firstName: 'Kyle',
            lastName: 'Bridgewater',
            githubUsername: 'kyleUnderwater',
            email: 'kylebridge@yahoo.com',
            id: 111
          },
          {
            firstName: 'Billy',
            lastName: 'Sanders',
            githubUsername: 'billysanders101',
            email: 'billysanders101@gmail.com',
            id: 112
          }
        ]
      },
      {
        cohortName: 'Cohort 10',
        students: [
          {
            firstName: 'Alexandra',
            lastName: "O'neil",
            githubUsername: 'zandOneil4',
            email: 'AlexOneil@gmail.com',
            id: 211
          }
        ]
      }
    ]
```

I have an array as the value of the students key, and that array is populated with objects containing the student data. There is also the cohort that they belong to shown in the parent object:

```
{
        cohortName: 'Cohort 9',
        students: [
          {
            firstName: 'Kyle',
            lastName: 'Bridgewater',
            githubUsername: 'kyleUnderwater',
            email: 'kylebridge@yahoo.com',
            id: 111
          },
          {
            firstName: 'Billy',
            lastName: 'Sanders',
            githubUsername: 'billysanders101',
            email: 'billysanders101@gmail.com',
            id: 112
          }
        ]
      }
```

File: src/cohortManager.js (hunk 5 above)
Comment (marc-alzapiedi) [reply src/cohortManager.js:27]: I think the student info was a requirement to add, including the studentID

**Pull request closed without merging**